unified-installer/astra-unified-installer.sh

```diff
@@ -1117,12 +1179,177 @@ is_astra_registry() {
+    [ ! -f "$kustomization_file" ] && fatal "kustomization file '$kustomization_file' does not exist"
+    [ -z "$namespace" ] && fatal "no namespace given"
+    [ -z "$registry" ] && fatal "no registry given"
+
+    local -r account_id="$ASTRA_ACCOUNT_ID"
+    local -r api_token="$ASTRA_API_TOKEN"
+    local -r pull_secret="$__ASTRA_REGISTRY_IMAGE_PULL_SECRET"
+    [ -z "$account_id" ] && fatal "no account_id found"
+    [ -z "$api_token" ] && fatal "no api_token found"
+    [ -z "$pull_secret" ] && fatal "no pull_secret found"
+
+    local -r encoded_creds=$(echo -n "$account_id:$api_token" | base64)
+
+    # Add secret generator
+    insert_into_file_after_pattern "$kustomization_file" "secretGenerator:" \
+'- name: "'"$pull_secret"'"
+  namespace: "'"$namespace"'"
+  type: kubernetes.io/dockerconfigjson
+  literals:
+  - |
+    .dockerconfigjson={
+      "auths": {
+        "'"$registry"'": {
+          "username": "'"$account_id"'",
+          "password": "'"$api_token"'",
+          "auth": "'"$encoded_creds"'"
+        }
+      }
+    }'
+    logdebug "$kustomization_file: added secret '$pull_secret' to namespace '$namespace'"
+}
+
+add_api_token_secret_to_kustomization() {
+    local -r kustomization_file="${1}"
+    local -r namespace="${2}"
+    [ -z "$kustomization_file" ] && fatal "no kustomization file given"
+    [ ! -f "$kustomization_file" ] && fatal "kustomization file '$kustomization_file' does not exist"
+    [ -z "$namespace" ] && fatal "no namespace given"
+
+    local -r api_token="$ASTRA_API_TOKEN"
+    local -r pull_secret="$__ASTRA_API_TOKEN_PULL_SECRET"
+    [ -z "$api_token" ] && fatal "no api_token found"
+    [ -z "$pull_secret" ] && fatal "no pull_secret found"
+
+    # Add API token secret
+    insert_into_file_after_pattern "$kustomization_file" "secretGenerator:" \
+"- name: \"${pull_secret}\"
+  namespace: \"${namespace}\"
+  literals:
+  - apiToken=\"${api_token}\""
+    logdebug "$kustomization_file: added secret '$pull_secret' to namespace '$namespace'"
+}
+
+add_imagepullsecret_patch_to_kustomization() {
+    local -r kustomization_file="${1}"
+    local -a pull_secrets=("${@:2}")
+
+    [ -z "$kustomization_file" ] && fatal "no kustomization file given"
+    [ ! -f "$kustomization_file" ] && fatal "kustomization file '$kustomization_file' does not exist"
+    [ "${#pull_secrets[@]}" -eq 0 ] && fatal "no pull secrets given"
+
+    local patch_entries=""
+    for secret in "${pull_secrets[@]}" ; do
+        is_empty "$secret" && fatal "one of the given secrets is empty"
+        patch_entries+="${__NEWLINE}        - name: \"${secret}\""
+    done
+
+    insert_into_file_after_pattern "$kustomization_file" "patches:" \
+'- target:
+    kind: Deployment
+  patch: |-
+    - op: replace
+      path: /spec/template/spec/imagePullSecrets
+      value:'"${patch_entries}"''
+    logdebug "$kustomization_file: added pull secret patch for [${pull_secrets[*]}]"
+}
+
+add_namespace_transformer_to_kustomization() {
```

Comment:
This used to be `step_kustomize_global_namespace_if_needed`; as with the others, it was made generic because both the connector and trident kustomization files need it.